main.go

```diff
@@ -109,35 +136,14 @@ func init() {
 	pflag.Bool("once", false, "Run Ansible Puller just once, then exit")
 	pflag.Bool("version", false, "Print the build version, then exit")
 
-	err := viper.ReadInConfig()
-	if err != nil {
-		logrus.Fatalf("fatal error in config file: %s", err)
-	}
-
-	err = viper.BindPFlags(pflag.CommandLine)
-	if err != nil {
-		logrus.Fatal("unable to bind configuration")
-	}
-
-	pflag.Parse()
-
-	logrus.SetOutput(os.Stdout)
-	if viper.GetBool("debug") {
-		logrus.SetLevel(logrus.DebugLevel)
-		promDebug.Set(1)
-	} else {
-		logrus.SetFormatter(&logrus.JSONFormatter{})
-	}
-
-	if viper.GetBool("start-disabled") {
-		ansibleDisable()
-	}
+	loadConfig()
 
-	hostname, err = os.Hostname()
-	if err != nil {
-		logrus.Fatal("Unable to detect hostname")
-	}
+	viper.WatchConfig()
```

Comment:
Should we watch the config after defining the callback function through `OnConfigChange`?